src/abstractdataframe/io.jl

```diff
@@ -92,16 +99,74 @@ function html_escape(cell::AbstractString)
     return cell
 end
 
-Base.show(io::IO, mime::MIME"text/html", df::AbstractDataFrame; summary::Bool=true) =
-    _show(io, mime, df, summary=summary)
+"""
+    show(io::IO, mime::MIME, df::AbstractDataFrame;
+         allrows::Bool = !get(io, :limit, true),
+         allcols::Bool = !get(io, :limit, true),
+         summary::Bool=true)
+
+Render a data frame to an I/O stream using `mime`.
+
+# Arguments
+- `io::IO`: The I/O stream to which `df` will be printed.
+- `mime::MIME`: supoorted mime types are: `"text/html"`, `"text/latex"`, `"text/csv"`,
+   `"text/tab-separated-values"`
+- `df::AbstractDataFrame`: The data frame to print.
+- `allrows::Bool `: supported only for HTML and LaTeX mime types;
+   whether to print all rows, rather than a default number defined in
+   `HTML_NROWS` and `LATEX_NROWS` constants. The default value can be overriden by
```

Comment:
I'd rather not define our custom `IOContext` property, especially if it's a temporary solution. Better use `get(io, :displaysize, HTML_DEFAULT)` (which is different from `displaysize(io)`). That's quite simple to choose the number of rows. For columns it's a bit more tricky: how hard do you think it would be to compute the width in characters needed to print a given number of columns?

Reply:
> how hard do you think it would be to compute the width in characters needed to print a given number of columns?

This is what I am not sure of (see the post I was writing in parallel). There are two issues:
* we use variable width characters when printing
* we actually do not have to fit the `:displaysize` in Jupyter Notebook: it is OK (and I think desirable) to accept a bit larger output as we support scrolling

Reply:
For these reasons I was hesitant to use `:displaysize` or `displaysize(io)` but rather to have something we can customize. But I admit that I am not sure what is the best design here.

Maybe we could simply disallow customization of these parameters if this is a temporary solution (so we use the defaults or print everything). I do not think people would complain about it as most of the users would not even know that they can change the defaults or care about it. You always can print the exact slice of your data frame by writing `show(io, df[my_rows, my_columns], allrows=true, allcols=true)` so this should not be a great problem.

Reply:
Yes that's only supposed to be a rough rule to choose the size. The point is that I don't think we're supposed to use `IOContext` for custom properties like that, especially with so generic names. Also they are propagated to `show` methods for elements, which really isn't what we want here (even if in practice it's not a problem).

Maybe better just hardcode the number of rows and columns for now, and not support any `IOContext` property that isn't really supposed to be used in Jupyter anyway. We'll see later whether we can find a better approach.

Reply:
Right. I will do this.